cmd/traffic/cmd/manager/service.go

```diff
@@ -49,19 +54,47 @@ func (wall) Now() time.Time {
 	return time.Now()
 }
 
-func NewManager(ctx context.Context) (*Manager, context.Context) {
+func getCloudConfig(ctx context.Context) (*rpc.AmbassadorCloudConfig, error) {
+	const proxyCertsPath = "/var/run/secrets/proxy_tls"
+
+	env := managerutil.GetEnv(ctx)
+	ret := &rpc.AmbassadorCloudConfig{Host: env.SystemAHost, Port: env.SystemAPort}
+	if _, err := os.Stat(proxyCertsPath); err != nil {
+		if os.IsNotExist(err) {
+			return ret, nil
+		}
+		return nil, fmt.Errorf("could not stat %s: %w", proxyCertsPath, err)
+	}
+	f, err := os.Open(path.Join(proxyCertsPath, "ca.crt"))
+	if err != nil {
```

Comment:
Why is it OK if the directory doesn't exist but not OK if the directory exists and the file is missing?

Reply:
if the directory isn't there that means the secret just isn't mounted. On the other hand if the directory is present but the file is missing, then the secret is misconfigured/lacks the correct keys. IMO this is the correct behavior

Reply:
Ok, that makes sense.